source/s3_client.c

```diff
@@ -336,6 +342,29 @@ PyObject *aws_py_s3_client_new(PyObject *self, PyObject *args) {
     s3_client->py_core = py_core;
     Py_INCREF(s3_client->py_core);
 
+    if (network_interface_names_py != Py_None) {
+        if (!PySequence_Check(network_interface_names_py)) {
+            PyErr_SetString(PyExc_TypeError, "Expected network_interface_names to be a sequence.");
+            goto cleanup;
+        }
+        Py_ssize_t listSize = PySequence_Size(network_interface_names_py);
```

Comment:
[docs say](https://docs.python.org/3/c-api/sequence.html#c.PySequence_Size)
> Returns ... -1 on failure

docs don't say whether a python exception was set in that case ... but [looking at the code](https://github.com/python/cpython/blob/376d66eb5080cf8076d767b0916c103463343963/Objects/abstract.c#L1609-L1610) seems like yeah there would be an exception